### PR TITLE
[ESI] [Cosim] Array decoding

### PR DESCRIPTION
**Pull request by teqdruid**
Adds decoding of Cap'nProto lists to rtl.array.

### CODE REVIEW

File: lib/Dialect/RTL/RTLTypes.cpp
Comment (teqdruid) [inline lib/Dialect/RTL/RTLTypes.cpp:241]: This change is a separate PR (#528) but necessary for this one. I won't merge this one until #528 merges.

**Review by mikeurbach — APPROVED** [review]
This is nice. A little less whimsical but the new helper looks good.

**Pull request merged**
